cli/src/plz/cli/configuration.py

```diff
@@ -83,41 +83,120 @@ class Configuration:
             Property('debug', type=bool, default=False),
             Property('docker_run_args', type=dict, default={}),
             Property('connection_info', type=dict, default={}),
+            Property('context_path', type=str, default='.'),
         ]
     }
 
-    CONFIGURATION_FILE = 'plz.config.json'
+    DEFAULT_CONFIGURATION_FILE_NAME = 'plz.config.json'
     MISSING_CONFIGURATION_FILE_ERROR = ValidationError(
-        f'You must create a {CONFIGURATION_FILE} file.')
+        f'You must create a {DEFAULT_CONFIGURATION_FILE_NAME} file, '
+        f'or specify a path to it with -c')
 
     @staticmethod
-    def load() -> 'Configuration':
-        default_configuration = Configuration.defaults(
-            Configuration.PROPERTIES)
-        try:
-            file_configuration = Configuration.from_file(
-                Configuration.CONFIGURATION_FILE, Configuration.PROPERTIES)
-        except FileNotFoundError:
+    def load(configuration_path: Optional[str] = None) -> 'Configuration':
+        config_file_name = Configuration._configuration_file_from_path(
+            configuration_path)
+
+        configuration = Configuration.defaults(Configuration.PROPERTIES)
+        configuration, user_level_config_was_read = \
+            Configuration._override_with_user_level_config(configuration)
+
+        configuration, file_config_was_read = \
+            Configuration._override_with_file_configs(
+                config_file_name, configuration,
+                plz_config_set_explicitly=configuration_path is not None)
```

Comment:
Rather than instructing the method to override, why not return a configuration and merge them all together later? It would make the methods simpler, and remove the need for `config_was_read`, instead just checking for `None`.